### ISSUE TITLE
High contrast theme is not very high contrast while debugging

### ISSUE THREAD

**Issue by mousetraps** [Microsoft/vscode#15852]
- VSCode Version: Code 1.7.2 (7ba55c5860b152d999dda59393ca3ebeb1b5c85f, 2016-11-21T22:06:37.428Z)
- OS Version: Windows_NT ia32 10.0.14931

![image](https://cloud.githubusercontent.com/assets/762848/20508344/d273cb54-b015-11e6-8b76-252d42d64b78.png)

![image](https://cloud.githubusercontent.com/assets/762848/20508365/f4d1810a-b015-11e6-8df4-b75e3a323826.png)

![image](https://cloud.githubusercontent.com/assets/762848/20508385/1edab980-b016-11e6-88f8-e4e44e03caf7.png)

**Comment by weinand** [comment]
@isidorn Is this a regression in 1.7.2 or was this already a problem in 1.7?

**Comment by ramya-rao-a** [comment]
The variables are not readable on the line where breakpoint is hit

![image](https://cloud.githubusercontent.com/assets/16890566/20992650/5a8f973a-bc9a-11e6-8641-1f2a404ac33c.png)

The boolean values have a dark blue color with grey brackground which again is not that readable

![image](https://cloud.githubusercontent.com/assets/16890566/20992695/afa19192-bc9a-11e6-8080-781ecb9b1a66.png)


@bgashler1 thoughts?

**Comment by bgashler1** [comment]
I'm not sure I've seen this before.  This is really bad.  Contrast should be a minimum ratio of 7:1 for our high-contrast theme to meet AAA accessibility.  I'll do some work on this tomorrow.

**Comment by bgashler1** [comment]
For the problem @ramya-rao-a found:
Just discovered that this is most likely caused by using `mix-blend-mode: luminosity;` css rule on the editor foreground text.  We had to do that to create the appearance of inverted text when it's selected in this theme, since we cannot directly change the color of the text section that sits in front of a selection (today there is no way to style the text color of a selection, as far as I know).

The problem seems to be that the editor foreground text doesn't blend nicely with the yellow background color of `debug-top-stack-frame-line` (shown on a paused breakpoint).

**Comment by bgashler1** [comment]
@isidorn is it possible to put a class only on selected text so that we could put `filter: invert(100%)` on that class instead of doing our really hacky `mix-blend-mode` on all text tokens?  That would fix this problem.

**Comment by bgashler1** [comment]
Actually, I was able to improve it a lot just by getting rid of the alpha.  But that's not to say we couldn't come up with a less-hacky solution later :)

![screen shot 2016-12-07 at 5 28 36 pm](https://cloud.githubusercontent.com/assets/11839736/20994100/2ff55006-bca3-11e6-85b2-4b08634213be.png)

I'll work on the rest tomorrow.

**Comment by ramya-rao-a** [comment]
Re-opening the issue as there is still work to be done

**Comment by isidorn** [comment]
@bgashler1 you do not have to do this in the last day of the sprint. It was always like this and is not a regression. Thus assigning to Brad and January. Just polish the shit out of high contrast debugging starting from next week :)

**Comment by Tyriar** [comment]
@bgashler1 FYI people don't get notifications on commits, I happened to see it because I looked at https://github.com/Microsoft/vscode/commits/master

**Comment by bgashler1** [comment]
There's a lot that needs to be improved on the high-contrast theme (but like Isidor said, we can do that in the January iteration).  However, in https://github.com/Microsoft/vscode/commit/2b288227be7fea5b2de4387071a6ecb18ee599ed I fixed an important regression that I found from yesterday where text selection was not showing up at all on a paused breakpoint.  Just FYI.

**Comment by isidorn** [comment]
@bgashler1 thanks for fixing the regression.

**Comment by isidorn** [comment]
Most of the issues touched here are fixed -> closing
If there are additional minor issues please file a new issue
